### PR TITLE
Fix db escape negative integers

### LINKED ISSUE #4973: Bug: Query builder escapes negative integers
**Describe the bug**
In #606, a bit of code was added that escapes negative values. However, I think negative values should not be escaped at all. For example, I would like to look back at negative dates using something like

```sql
WHERE date_pickup < DateAdd(month, ?, Convert(date, GetDate())
```

Due the conversion, this is translated to

```sql
WHERE date_pickup < DateAdd(month, '-6', Convert(date, GetDate())
```
Which results in an error, since we now pass a varchar to the function instead of an integer.
My question is: why are negative values escaped at all? The offending bit of code checks that the integer is negative before escaping it, and I'm not sure what the original commit intends to solve.

**CodeIgniter 4 version**
Tested in 4.1.3, should be present in all versions.

**Affected module(s)**
`Database/BaseConnection.php` lines 1312 - 1314 in the escape function. Specifically when using SQLSRV driver. Though it should be independent of driver used.

**Context**
 - OS: Windows 10
 - Web server IIS
 - PHP version 8.0

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Fixes #4973

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Database/BaseConnection.php [hunk 1]
```diff
@@ -1201,7 +1203,7 @@ public function escape($str)
         }
 
         if (is_numeric($str) && $str < 0) {
```
Comment (paulbalandan) [inline system/Database/BaseConnection.php:1205]: Why is there a need to add this for negative numbers? I think we can remove this if block entirely as the last return will do the job.

Comment (paulbalandan) [reply system/Database/BaseConnection.php:1205]: And it seems inconsistent, with negative numbers we get a string. For positive, we can numbers.

Comment (kenjis) [reply system/Database/BaseConnection.php]: Good point!

For now, I remove it.

Comment (kenjis) [reply system/Database/BaseConnection.php]: However, I am not sure why this commit https://github.com/codeigniter4/CodeIgniter4/commit/19c173a49f35388d90736b6ee596f33a5f29a870 was fixed #606.

Comment (paulbalandan) [reply system/Database/BaseConnection.php]: let's have @lonnieezell shed the light on this. 😂

Comment (lonnieezell) [reply system/Database/BaseConnection.php]: I don't remember exactly. It's been 4 years :) . Looking back through the issue, though, I'm guessing that wrapping it in quotes would get it past the escaping and adding of the slash, which was then converted by MySQL into an INT or whatever type when it was saved to the table. 

I would say as long as we have tests in place that ensure his problem is still fixed, and it doesn't escape the negative sign, this is fine.

File: tests/system/Database/Live/EscapeTest.php [hunk 2]
```diff
@@ -40,7 +40,7 @@ protected function setUp(): void
      */
     public function testEscapeProtectsNegativeNumbers()
```
Comment (paulbalandan) [inline tests/system/Database/Live/EscapeTest.php:41]: This should be renamed to say that this does not escape negative numbers now.

Comment (kenjis) [reply tests/system/Database/Live/EscapeTest.php]: Fixed.

File: system/Database/BaseConnection.php (hunk 1 above)
Comment (kenjis) [reply system/Database/BaseConnection.php]: @lonnieezell Thanks!

**Review by paulbalandan — APPROVED** [review]

**Pull request merged**